platform/ios/src/MGLMapView.h

```diff
@@ -45,6 +45,31 @@ typedef NS_ENUM(NSUInteger, MGLAnnotationVerticalAlignment) {
 };
 
 /**
+ The mode used to track the user location on the map. Used with
+ `MGLMapView.userTrackingMode`.
+ */
+typedef NS_ENUM(NSUInteger, MGLUserTrackingMode) {
+    /** The map does not follow the user location. */
+    MGLUserTrackingModeNone              = 0,
+    /** The map follows the user location. */
+    MGLUserTrackingModeFollow,
+    /** The map follows the user location and rotates when the heading changes.
+        The default user location annotation shows a fan-shaped indicator with
+        the current heading, sized according to the reported accuracy. This
+        tracking mode is disabled if the user pans the map view, but remains
+        unchanged if the user zooms in. If the user rotates the map view, this
+        tracking mode will fall back to `MGLUserTrackingModeFollow`. */
+    MGLUserTrackingModeFollowWithHeading,
+    /** The map follows the user location and rotates when the course changes.
+        The default user location annotation shows a puck-shaped indicator
+        that rotates as the user course changes. This tracking mode is
+        disabled if the user pans the map view, but remains if the user zooms
+        in. If the user rotates the map view, this tracking mode will fall back
+        to `MGLUserTrackingModeFollow`. */
```

Comment:
Maybe we should refer the developer to `CLLocation.course`, since the distinction between a heading and a course is often lost on developers.